internal/fwschemadata/data_set.go

```diff
@@ -10,11 +10,17 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/internal/reflect"
 	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
 )
 
 // Set replaces the entire value. The value should be a struct whose fields
 // have one of the attr.Value types. Each field must have the tfsdk field tag.
 func (d *Data) Set(ctx context.Context, val any) diag.Diagnostics {
+	if v, ok := val.(tftypes.Value); ok {
+		d.TerraformValue = v
+		return nil
+	}
```

Comment:
We probably want to check that the supplied `val` actually corresponds to the schema of the object we're setting this too?

Reply:
This has been moved over into https://github.com/hashicorp/terraform-plugin-framework/pull/1198 since it's related to that piece of work